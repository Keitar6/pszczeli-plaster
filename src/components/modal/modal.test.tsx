import { createElement, ReactNode, ReactPortal } from "react";
import ReactDOM from "react-dom";
import ComponentsRenderer from "react-test-renderer";
import Modal from "./modal.component";

describe("modal test", () => {
  const prevPortal = ReactDOM.createPortal;
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((node: ReactNode) => {
      return node as ReactPortal;
    });
  });
  test("portal test", () => {
    const component = ComponentsRenderer.create(<Modal>Hi</Modal>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  afterAll(() => {
    ReactDOM.createPortal = prevPortal;
  });
});
