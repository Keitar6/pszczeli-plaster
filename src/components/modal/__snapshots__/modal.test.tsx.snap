// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal test portal test 1`] = `"Hi"`;
